server/base/src/main/java/org/apache/accumulo/server/fs/VolumeUtil.java

```diff
@@ -275,10 +274,10 @@ private static String decommisionedTabletDir(ServerContext context, ZooLock zooL
 
     VolumeChooserEnvironment chooserEnv = new VolumeChooserEnvironment(extent.getTableId(),
         context);
-    Path newDir = new Path(
-        vm.choose(chooserEnv, ServerConstants.getBaseUris(context.getConfiguration()))
-            + Path.SEPARATOR + ServerConstants.TABLE_DIR + Path.SEPARATOR
-            + dir.getParent().getName() + Path.SEPARATOR + dir.getName());
+    Path newDir = new Path(vm.choose(chooserEnv,
+        ServerConstants.getBaseUris(context.getConfiguration(), context.getHadoopConf()))
```

Comment:
Could only context be passed here?

Reply:
In some cases yes but we still use this in Initialize where we don't have a good way to create ServerContext yet.

Reply:
Ok..  Could possibly have two methods.  One that takes context and one that takes everything.  This would simplify most code while leaving the init code as is.